### PR TITLE
New ui media app

### PR DESCRIPTION
**Pull request by shahidajmeri786**
1. Added filter type feature
2. Added trash features for extras
3. Added detail view of media with action buttons

### CODE REVIEW

File: backend/src/Controllers/AttachmentsController.php [hunk 1]
```diff
@@ -163,13 +185,37 @@ public function count( WP_REST_Request $request ) {
...
+	{
 
 		$args = $request->get_params();
 
-		return $this->attachments->paginate( $args )
-								 ->apply_transform( $this->transformer )
-								 ->to_rest_response();
+		switch ($args['post_mime_type']) {
+			case 'spreadsheets':
+				$args['post_mime_type'] = 'pplication/vnd.apple.numbers,application/vnd.oasis.opendocument.spreadsheet,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,application/vnd.ms-excel.sheet.macroEnabled.12,application/vnd.ms-excel.sheet.binary.macroEnabled.12';
```
Comment (brentjett) [inline backend/src/Controllers/AttachmentsController.php]: might be a missing "a" here: `= 'pplication/...`

File: src/system/ui/mediaupload/index.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import { http } from 'utils/wordpress'
+import { Icon } from 'ui'
+import { createStoreRegistry } from 'fluid/data'
```
Comment (brentjett) [inline src/system/ui/mediaupload/index.js]: You don't actually need a new store registry here, but it looks like `registerStore` wasn't exported from the default store registry. I've added that to `new-ui` now so you should be able to simply `import { registerStore } from 'data'`.

File: src/system/ui/pages/attachment/index.js [hunk 3]
```diff
@@ -37,7 +92,19 @@ export const Attachment = ( { location } ) => {
 				description: {
 					label: __( 'Description' ),
 					component: 'textarea',
-					rows: 4,
```
Comment (brentjett) [inline src/system/ui/pages/attachment/index.js]: I think i changed this to 4 on purpose.

File: src/system/utils/wordpress/rest.js [hunk 4]
```diff
@@ -43,7 +44,7 @@ const cache = setupCache( {
  *
  * @type {AxiosInstance}
  */
-const http = axios.create( {
+export const http = axios.create( {
```
Comment (brentjett) [inline src/system/utils/wordpress/rest.js]: Does this really need to be exported? That's adding it to the public api by doing that.

File: src/system/ui/mediaupload/index.js (hunk 2 above)
Comment (shahidajmeri786) [reply src/system/ui/mediaupload/index.js]: I took a pull of new-ui and merged with media app and tried to import with `import { registerStore }` but getting error.
image

File: src/system/ui/pages/attachment/index.js (hunk 3 above)
Comment (shahidajmeri786) [reply src/system/ui/pages/attachment/index.js]: Updated.

File: backend/src/Controllers/AttachmentsController.php (hunk 1 above)
Comment (shahidajmeri786) [reply backend/src/Controllers/AttachmentsController.php]: Added missed 'a'

File: src/system/utils/wordpress/rest.js (hunk 4 above)
Comment (shahidajmeri786) [reply src/system/utils/wordpress/rest.js]: Not needed anymore. Used create method of attachment from rest.js

**Comment by fastlinemedia** [comment]
Thanks, Shahid. I just reviewed the frontend, haven't looked at the code yet. Just a few things...

1) When I make changes and save them, they are not showing up in wp-admin. The old data is still showing. So it looks like changes might not be saving to the DB.

2) Can we put [this stuff](https://share.getcloudapp.com/Z4uwg4yB) in a section called "Info" with the labels besides the text content? 

3) Can you remove the [not hooked up yet item](https://share.getcloudapp.com/2Nur4DPP)?

File: backend/src/Controllers/AttachmentsController.php [hunk 5]
```diff
@@ -31,57 +32,76 @@ class AttachmentsController extends ControllerAbstract {
...
+					'permission_callback' => function () {
+						return current_user_can('edit_published_posts');
+					},
+				],
+
+			]
+		);
+
+		$this->route(
+			'/attachments/upload',
```
Comment (fastlinemedia) [inline backend/src/Controllers/AttachmentsController.php]: @shahidajmeri786 do we still need this route and the related method? It looks like it's not being used.

Comment (shahidajmeri786) [reply backend/src/Controllers/AttachmentsController.php]: Removed.

**Comment by shahidajmeri786** [comment]
> Thanks, Shahid. I just reviewed the frontend, haven't looked at the code yet. Just a few things...
> 
> 1. When I make changes and save them, they are not showing up in wp-admin. The old data is still showing. So it looks like changes might not be saving to the DB.
> 2. Can we put [this stuff](https://share.getcloudapp.com/Z4uwg4yB) in a section called "Info" with the labels besides the text content?
> 3. Can you remove the [not hooked up yet item](https://share.getcloudapp.com/2Nur4DPP)?



> Thanks, Shahid. I just reviewed the frontend, haven't looked at the code yet. Just a few things...
> 
> 1. When I make changes and save them, they are not showing up in wp-admin. The old data is still showing. So it looks like changes might not be saving to the DB.
> 2. Can we put [this stuff](https://share.getcloudapp.com/Z4uwg4yB) in a section called "Info" with the labels besides the text content?
> 3. Can you remove the [not hooked up yet item](https://share.getcloudapp.com/2Nur4DPP)?

Resolved above mentioned issues.

**Pull request merged**